src/components/common/promotioncard.tsx

```diff
@@ -15,17 +15,25 @@ const PromotionCard: FC<Props> = ({ promotion }) => {
       header={
         <div className="flex items-center px-4 pt-4 pb-2 border-solid border-b border-b-[#383F47]">
           <CustomImage src={CheckCircle} alt="promotion" />
-          <span className="text-[#faf9f6] text-sm pl-4 uppercase font-bold text-base">
+          <span
+            className="text-[#faf9f6] text-sm pl-4 uppercase font-bold text-base whitespace-nowrap text-ellipsis overflow-hidden w-[15rem] sm:w-[20rem]"
+            title={promotion.claimableId}
```

Comment:
why title attribute for span ?

Reply:
I've added ellipses and tooltip to handle text overflow, and I got confirmation from Vyshak as well.